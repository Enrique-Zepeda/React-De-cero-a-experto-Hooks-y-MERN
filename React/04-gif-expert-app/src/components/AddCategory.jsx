import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");
  const onInputChange = (e) => {
    // console.log(e.target.value);
    setInputValue(e.target.value);
  };

  const onSubmit = (e) => {
    e.preventDefault();
    console.log(inputValue);
    //Es para que no se pueda mandar datos vacios
    if (inputValue.trim().length <= 1) return;
    // Esto lo hice mandandole tambien las categorias por prop pero se puede evitar
    // setCategories([inputValue, ...categories]);
    //Asi se usa la funcion sin depender del prop de categorias
    // setCategories((categories) => [inputValue, ...categories]);

    //esto es solo mandanlo los props la padre
    onNewCategory(inputValue.trim());
    setInputValue("");
  };
  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
