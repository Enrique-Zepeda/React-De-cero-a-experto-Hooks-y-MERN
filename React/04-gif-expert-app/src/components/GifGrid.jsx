import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";
import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {
  const [imagenes, setImagenes] = useState([]);
  const getImage = async () => {
    const newImagenes = await getGifs(category);
    setImagenes(newImagenes);
    console.log("Este es el estado chido", newImagenes);
  };

  useEffect(() => {
    getGifs(category);
    getImage();
  }, []);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {imagenes.map((image) => (
          <GifItem
            key={image.id}
            //sirve para poder tomas las propiedades
            {...image}
            //   title={image.title}
            //   url={image.url}
          />
        ))}
      </div>
    </>
  );
};
