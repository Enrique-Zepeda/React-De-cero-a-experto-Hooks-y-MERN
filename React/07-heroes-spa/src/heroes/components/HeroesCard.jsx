import { Link } from "react-router-dom";

const CharactersByHero = ({ alter_ego, characters }) => {
  if (alter_ego === characters) return <></>;

  return <p>{characters}</p>;
};

export const HeroesCard = ({ hero }) => {
  const heroUrl = `/assets/heroes/${hero.id}.jpg`;

  // const characters = ( <p>{hero.characters}</p>)

  return (
    <div className="col">
      <div className="card">
        <div className="row no-gutters">
          <div className="col-4">
            <img src={heroUrl} className="card-img" alt={hero.superhero} />
          </div>

          <div className="col-8">
            <div className="car-body">
              <h5 className="card-title">{hero.superhero}</h5>
              <p className="card-text">{hero.alter_ego}</p>

              {/* <p className="card-text">{hero.characters}</p> */}
              <CharactersByHero
                characters={hero.characters}
                alter_ego={hero.alter_ego}
              />

              <p className="card-text">
                <small className="text-muted">{hero.first_appearance}</small>
              </p>

              <Link to={`/hero/${hero.id}`}>Mas...</Link>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
