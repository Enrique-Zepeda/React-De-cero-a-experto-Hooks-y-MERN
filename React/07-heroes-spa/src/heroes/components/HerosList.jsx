import { getHeroesByPublisher } from "../helpers/getHeroesByPublisher";

export const HerosList = ({ publisher }) => {
  const heroes = getHeroesByPublisher(publisher);
  //   console.log(heroes);

  return (
    <ul>
      {heroes.map((hero) => (
        <li key={hero.id}>{hero.characters}</li>
      ))}
    </ul>
  );
};
