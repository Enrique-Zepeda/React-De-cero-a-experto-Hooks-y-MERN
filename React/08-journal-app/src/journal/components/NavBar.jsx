import { AppBar, Grid, IconButton, Toolbar, Typography } from "@mui/material";
import { LoginOutlined, MenuOutlined } from "@mui/icons-material";
import { useDispatch } from "react-redux";
import { startLogout } from "../../store/auth/thunks";

export const Navbar = ({ drawerWidth }) => {
  const dispatch = useDispatch();

  const onLogout = () => {
    // console.log("logOut");
    dispatch(startLogout());
  };

  return (
    <AppBar
      position="fixed"
      sx={{
        width: { sm: `calc(100% - ${drawerWidth}px)` },
        ml: { sm: `${drawerWidth}px` },
      }}
    >
      <Toolbar>
        <IconButton color="inherit" sx={{ mr: 2, display: { sm: "none" } }}>
          <MenuOutlined />
        </IconButton>

        <Grid
          container
          sx={{ width: "100%" }}
          display="flex"
          direction="row"
          justifyContent="space-between"
          alignItems="center"
        >
          <Typography variant="h6" noWrap component="div">
            JournalApp
          </Typography>
          <IconButton color="error" onClick={onLogout}>
            <LoginOutlined />
          </IconButton>
        </Grid>
      </Toolbar>
    </AppBar>
  );
};
