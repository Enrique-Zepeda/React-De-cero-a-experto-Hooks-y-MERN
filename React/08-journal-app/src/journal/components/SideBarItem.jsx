import {
  ListItem,
  ListItemButton,
  ListItemIcon,
  ListItemText,
  Grid,
} from "@mui/material";
import { TurnedInNot } from "@mui/icons-material";
import { useMemo } from "react";
import { useDispatch } from "react-redux";
import { setActiveNote } from "../../store/journal/journalSlice";

export const SideBarItem = ({ title, id, body, date, imageUrls = [] }) => {
  const dispatch = useDispatch();

  const newTitle = useMemo(() => {
    return title.length > 17 ? title.substring(0, 17) + "..." : title;
  }, [title]);

  const activeNote = () => {
    dispatch(
      setActiveNote({
        id,
        title,
        body,
        date,
        imageUrls,
      })
    );
  };

  return (
    <ListItem disablePadding>
      <ListItemButton onClick={activeNote}>
        <ListItemIcon>
          <TurnedInNot />
        </ListItemIcon>
        <Grid container>
          <ListItemText primary={newTitle} />
          <ListItemText secondary={body} />
        </Grid>
      </ListItemButton>
    </ListItem>
  );
};
