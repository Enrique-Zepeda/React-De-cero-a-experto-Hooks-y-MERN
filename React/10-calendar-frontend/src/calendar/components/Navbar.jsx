import { useState } from "react";
import { useAuthStore } from "../../hooks";

export const Navbar = () => {
  const { startLogout, user } = useAuthStore();
  const [animate, setAnimate] = useState("");
  const logOut = () => {
    setAnimate("animate__animated animate__fadeOutDown");

    setTimeout(() => {
      startLogout();
    }, 1000);
  };

  return (
    <div className={`navbar navbar-dark bg-dark mb-4 px-4 ${animate}`}>
      <span className="navbar-brand">
        <i className="fas fa-calendar-alt m-2"></i>
        {user.name}
      </span>
      <button className="btn btn-outline-danger" onClick={logOut}>
        <i className="fas fa-sign-out-alt m-2"></i>
        &nbsp;
        <span>Salir</span>
      </button>
    </div>
  );
};
