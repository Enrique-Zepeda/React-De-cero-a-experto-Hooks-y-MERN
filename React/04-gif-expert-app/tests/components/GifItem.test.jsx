import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";
describe("Pruebas en GifItem", () => {
  test("Probando snapshot", () => {
    const { container } = render(
      <GifItem title="titulo" url="https://www.algo.com" />
    );
    expect(container).toMatchSnapshot();
  });
});
