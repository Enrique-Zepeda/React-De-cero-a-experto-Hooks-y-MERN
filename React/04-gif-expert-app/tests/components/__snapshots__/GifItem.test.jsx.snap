// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Probando snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="titulo"
      src="https://www.algo.com"
    />
    <p>
      titulo
    </p>
  </div>
</div>
`;
